src/ChainSafe.GamingSdk.Gelato/dto/CallWithErc2771Request.cs

```diff
@@ -0,0 +1,208 @@
+        /// </summary>
+        CallWithSyncFee,
+
+        /// <summary>
+        /// For sponsoring the transaction from the developers account
+        /// </summary>
+        SponsoredCall,
+    }
+
+    [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
+
+    public class CallWithErc2771Request
+    {
+        /// <summary>
+        ///    QUANTITY - The transaction chain id.
+        /// </summary>
+        [JsonProperty(PropertyName = "chainId")]
+        public int ChainId { get; set; }
+
+        /// <summary>
+        ///     DATA, 20 Bytes - The address the transaction is being sent to.
+        /// </summary>
+        [JsonProperty(PropertyName = "target")]
+        public string Target { get; set; }
+
+        /// <summary>
+        ///     DATA - the data send along with the transaction.
+        /// </summary>
+        [JsonProperty(PropertyName = "data")]
+        public string Data { get; set; }
+
+        /// <summary>
+        ///     DATA, 20 Bytes - the address of the token that is to be used for payment.
+        /// </summary>
+        [JsonProperty(PropertyName = "feeToken")]
+        public string FeeToken { get; set; }
+
+        /// <summary>
+        ///     DATA - an optional boolean (default: true ) denoting what data you would prefer appended to the end of the calldata.
+        /// </summary>
+        [JsonProperty(PropertyName = "isRelayContext")]
+        public bool IsRelayContext { get; set; }
+
+        /// <summary>
+        ///     DATA, 20 Bytes - the address of the user's EOA.
+        /// </summary>
+        [JsonProperty(PropertyName = "user")]
+        public string User { get; set; }
+
+        /// <summary>
+        ///    QUANTITY - optional, this is a nonce similar to Ethereum nonces, stored in a local mapping on the relay contracts. It is used to enforce nonce ordering of relay calls, if the user requires it. Otherwise, this is an optional parameter and if not passed, the relay-SDK will automatically query on-chain for the current value.
+        /// </summary>
+        [JsonProperty(PropertyName = "userNonce")]
+        public int? UserNonce { get; set; }
+
+        /// <summary>
+        ///    QUANTITY - optional, the amount of time in seconds that a user is willing for the relay call to be active in the relay backend before it is dismissed.
+        /// </summary>
+        [JsonProperty(PropertyName = "userDeadline")]
+        public int? UserDeadline { get; set; }
+
+        /// <summary>
+        ///    DATA - the signature from the sign typed data request.
+        /// </summary>
+        [JsonProperty(PropertyName = "userSignature")]
+        public string Signature { get; set; }
+
+        /// <summary>
+        ///    DATA - the signature from the sign typed data request.
+        /// </summary>
+        [JsonProperty(PropertyName = "sponsorApiKey")]
+        public string SponsorApiKey { get; set; }
+
+        public TStructType MapRequestToStruct<TStructType>(
```

Comment:
I would move this out of the DTO class to some extension class, but that's discussable.

Reply:
There isn't a wide scope of usage for this file type in terms of going beyond its current use, Extensions would imply a multitude of usage options where this is pretty static in how its interacted with.

With that in mind, let me know if I should create the extension class

Reply:
Again, would be great to hear someone's else opinion, but I just don't like the idea of DTOs containing logic, that's larger than a single-line property. @Arshia001

Reply:
If I go into detail on what exactly I think about this matter, it will devolve into a philosophical discussion around why Object Oriented Programming was the worst idea we as an industry could have had. So, I'll keep it simple and just say that DTO-related logic in DTO classes isn't necessarily a bad thing, so let's keep it.

Reply:
This is used to transform the DTO object from standard class structure, to the specific Struct object that is needed for the sign typed data request. Both are DTO's, of the same data, for different steps in its consumption.